### DISCUSSION TITLE
URL not being updated after middleware redirect

### DISCUSSION THREAD

**Discussion by ghost** [livewire/livewire#7494] [7. Bugs (archived)]
### Livewire version

3.2.6

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Livewire is not updating the URL after redirect via auth middleware.

![Untitled](https://github.com/livewire/livewire/assets/62476094/014a0692-3c4a-4859-87b8-38fd30264b54)

Steps:

1. Visit home page
2. Click link to the protected page
3. Auth middleware redirects to login page but the URL is `example.com/protected` insted of `example.com/login`

Packages:
- Laravel
- Folio
- Livewire
- Volt

HOME PAGE

```
<?php

Laravel\Folio\name('home');

?>

<x-layouts.app>
    @volt
        <div>
            <h1>index.blade.php</h1>

            <a href="{{ route('protected') }}" wire:navigate>
                Link to protected route with "auth" middleware
            </a>
        </div>
    @endvolt
</x-layouts.app>
```

PROTECTED PAGE

```
<?php

Laravel\Folio\name('protected');
Laravel\Folio\middleware('auth');

?>

<x-layouts.app>
    @volt
        <div>
            <h1>protected.blade.php</h1>
        </div>
    @endvolt
</x-layouts.app>
```

LOGIN PAGE

```
<?php

Laravel\Folio\name('login');

?>

<x-layouts.app>
    @volt
        <div>
            <h1>login.blade.php</h1>
        </div>
    @endvolt
</x-layouts.app>
```

### Additional information

Related discussion? [https://github.com/livewire/livewire/discussions/4013](https://github.com/livewire/livewire/discussions/4013)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by mikerockett** [comment]
Same problem here, but in the reverse. I am applying the `RedirectIfAuthenticated` (guest) middleware to the login route. Livewire renders the target route (as redirected) properly, but the route itself does not change.
